### PR TITLE
[MM-47667] Bugfix: Cannot add users from Admin Console

### PR DESCRIPTION
**Pull request by plant99**
#### Summary

This PR fixes a bug where `this.state` was directly being modified while adding users from admin console.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-47667

#### Release Note

```release-note
Fixed a bug where users could not be added from the system console.
```

### CODE REVIEW

File: components/admin_console/team_channel_settings/team/details/team_details.tsx [hunk 1]
```diff
@@ -339,15 +339,16 @@ export default class TeamDetails extends React.PureComponent<Props, State> {
     addUsersToAdd = (users: UserProfile[]) => {
         let {usersToRemoveCount} = this.state;
         const {usersToAdd, usersToRemove} = this.state;
+        const usersToAddCopy = cloneDeep(usersToAdd);
         users.forEach((user) => {
             if (usersToRemove[user.id]?.id === user.id) {
                 delete usersToRemove[user.id];
                 usersToRemoveCount -= 1;
```
Comment (BenCookie95) [inline components/admin_console/team_channel_settings/team/details/team_details.tsx:346]: We are also modifying state directly here, should we also fix this?

**Review by mkraft — APPROVED** [review]

File: components/admin_console/team_channel_settings/team/details/team_details.tsx (hunk 1 above)
Comment (plant99) [reply components/admin_console/team_channel_settings/team/details/team_details.tsx:346]: Thanks for raising this Ben.

So this was inspired from https://stackoverflow.com/a/55567414 , and I could see that other appends i.e to `usersToRemove` work just fine.

Surely this doesn't fix the root cause of the issue here, I think. I'll keep looking.

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working fine

**Before**
image

**After**
image

**Pull request merged**